### PR TITLE
Handle JSON event query responses that contain a nil result field

### PR DESCRIPTION
**Pull request by frozenbubbleboy**

### CODE REVIEW

File: lib/zenoss/jsonapi.rb [hunk 1]
```diff
@@ -63,6 +63,13 @@ def parse_json(resp)
         end
 
         json = JSON.load(resp.http_body.content)
+        # Handle the situation where the 'result' key in the JSON response does not
+        # point to a hash, but instead is nil.
+        # This has been seen in the wild on an installation of Zenoss 4.2.5
+        if(json['result'].nil?
```
Comment (bjmllr) [inline lib/zenoss/jsonapi.rb]: Extra `(`?

**Comment by frozenbubbleboy** [comment]
Yeah, sorry about that.  I've removed the extra (.

**Comment by bjmllr** [comment]
The Travis failures were because we needed to update a couple things. I've done this in #23, can you rebase on to master and try pushing again when you get a chance?

**Comment by bjmllr** [comment]
Based on our discussion, it sounds like this isn't absolutely needed. This also might be a breaking change for someone who handles these `nil` responses in their app. I'll close this for now.

**Pull request closed without merging**